pkg/compare/compare.go

```diff
@@ -0,0 +1,118 @@
+// Copyright 2019 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package compare
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/google/go-cmp/cmp"
+
+	"sigs.k8s.io/yaml"
+
+	"istio.io/operator/pkg/tpath"
+)
+
+// YAMLCmpReporter is an option to generate tree based diff for YAMLs, used by cmp.Equal().
```

Comment:
this comment is a bit confusing... it doesn't look like this is an option.

Reply:
Yes, it is a reporter. Updated.